dask/dataframe/io/parquet.py

```diff
@@ -268,50 +334,150 @@ def _read_fastparquet(fs, fs_token, paths, columns=None, filters=None,
                        pf.file_scheme, storage_name_mapping)
-           for i, rg in enumerate(rgs)}
-    if not dsk:
-        # empty dataframe
-        dsk = {(name, 0): meta}
-        divisions = (None, None)
-        return out_type(dsk, name, meta, divisions)
+           for i, path in enumerate(paths)}
+    divisions = (None, ) * (len(paths) + 1)
+    return out_type(dsk, name, meta, divisions)
 
-    if index_names and infer_divisions is not False:
-        index_name = meta.index.name
-        try:
-            # is https://github.com/dask/fastparquet/pull/371 available in
-            # current fastparquet installation?
-            minmax = fastparquet.api.sorted_partitioned_columns(pf, filters)
-        except TypeError:
-            minmax = fastparquet.api.sorted_partitioned_columns(pf)
-        if index_name in minmax:
-            divisions = minmax[index_name]
-            divisions = divisions['min'] + [divisions['max'][-1]]
-        else:
-            if infer_divisions is True:
-                raise ValueError(
-                    ("Unable to infer divisions for index of '{index_name}' because it is not known to be "
-                     "sorted across partitions").format(index_name=index_name))
 
-            divisions = (None,) * (len(rgs) + 1)
+def _read_pf_simple(fs, path, base, index_names, all_columns, is_series,
+                    categories, cats, scheme, storage_name_mapping):
+    """Read dataset with fastparquet using ParquetFile machinery"""
+    from fastparquet import ParquetFile
+    pf = ParquetFile(path, open_with=fs.open)
+    relpath = path.replace(base, '').lstrip('/')
+    for rg in pf.row_groups:
+        for ch in rg.columns:
+            ch.file_path = relpath
+    pf.file_scheme = scheme
+    pf.cats = cats
+    pf.fn = base
+    df = pf.to_pandas(all_columns, categories, index=index_names)
+    if df.index.nlevels == 1:
+        if index_names:
+            df.index.name = storage_name_mapping.get(index_names[0],
+                                                     index_names[0])
     else:
-        if infer_divisions is True:
-            raise ValueError(
-                'Unable to infer divisions for because no index column was discovered')
+        if index_names:
+            df.index.names = [storage_name_mapping.get(name, name)
+                              for name in index_names]
+    df.columns = [storage_name_mapping.get(col, col)
+                  for col in all_columns
+                  if col not in (index_names or [])]
 
-        divisions = (None,) * (len(rgs) + 1)
+    if is_series:
+        return df[df.columns[0]]
+    else:
+        return df
 
-    if isinstance(divisions[0], np.datetime64):
-        divisions = [pd.Timestamp(d) for d in divisions]
 
-    return out_type(dsk, name, meta, divisions)
+def _paths_to_cats(paths, scheme):
+    """Extract out fields and labels from directory names"""
+    # can be factored out in fastparquet
```

Comment:
What does this comment mean? Is there similar code in fastparquet?

Reply:
Yes, but it's tied into a class method, and I don't have the appetite to change it and put in a set of deprivations.